test/e2e/page-objects/pages/account-list-page.ts

```diff
@@ -5,60 +5,68 @@ class AccountListPage {
 
   private accountListItem: string;
 
+  private accountMenuButton: string;
+
+  private accountNameInput: string;
+
   private accountOptionsMenuButton: string;
 
-  private hideUnhideAccountButton: string;
+  private addAccountConfirmButton: string;
 
-  private hiddenAccountsList: string;
+  private addEthereumAccountButton: string;
 
-  private hiddenAccountOptionsMenuButton: string;
+  private addSnapAccountButton: object;
 
-  private pinnedIcon: string;
+  private closeAccountModalButton: string;
 
-  private pinUnpinAccountButton: string;
+  private closeEditLabelButton: string;
 
   private createAccountButton: string;
 
-  private addEthereumAccountButton: string;
+  private editableLabelButton: string;
 
-  private accountNameInput: string;
+  private editableLabelInput: string;
 
-  private addAccountConfirmButton: string;
+  private hideUnhideAccountButton: string;
 
-  private accountMenuButton: string;
+  private hiddenAccountOptionsMenuButton: string;
 
-  private editableLabelButton: string;
+  private hiddenAccountsList: string;
 
-  private editableLabelInput: string;
+  private pinUnpinAccountButton: string;
 
-  private saveAccountLabelButton: string;
+  private pinnedIcon: string;
 
-  private closeEditLabelButton: string;
+  private saveAccountLabelButton: string;
 
   constructor(driver: Driver) {
     this.driver = driver;
+    this.accountListItem = '.multichain-account-menu-popover__list--menu-item';
+    this.accountMenuButton = '[data-testid="account-list-menu-details"]';
+    this.accountNameInput = '#account-name';
     this.accountOptionsMenuButton =
       '[data-testid="account-list-item-menu-button"]';
-    this.hideUnhideAccountButton = '[data-testid="account-list-menu-hide"]';
-    this.pinUnpinAccountButton = '[data-testid="account-list-menu-pin"]';
-    this.hiddenAccountsList = '[data-testid="hidden-accounts-list"]';
-    this.pinnedIcon = '[data-testid="account-pinned-icon"]';
-    this.hiddenAccountOptionsMenuButton =
-      '.multichain-account-menu-popover__list--menu-item-hidden-account [data-testid="account-list-item-menu-button"]';
-    this.createAccountButton =
-      '[data-testid="multichain-account-menu-popover-action-button"]';
-    this.addEthereumAccountButton =
-      '[data-testid="multichain-account-menu-popover-add-account"]';
-    this.accountNameInput = '#account-name';
     this.addAccountConfirmButton =
       '[data-testid="submit-add-account-with-name"]';
-    this.accountMenuButton = '[data-testid="account-list-menu-details"]';
+    this.addEthereumAccountButton =
+      '[data-testid="multichain-account-menu-popover-add-account"]';
+    this.addSnapAccountButton = {
+      text: 'Add account Snap',
+      tag: 'button',
+    };
+    this.closeAccountModalButton = 'button[aria-label="Close"]';
+    this.closeEditLabelButton = 'button[aria-label="Close"]';
```

Comment:
these two selectors are the same, I'm thinking if we can find a better selector for both, maybe using a test-id (if it's not too complex to add)

Reply:
oh, nice catch, thanks! I found that the close button for modals on the account list is the same, and since we don't have case where multiple modals are open on the account list, I will keep only one selector for the close button on different modals in account list.